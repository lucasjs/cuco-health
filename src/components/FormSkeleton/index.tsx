import React from 'react'
import { Skeleton, SimpleGrid } from '@chakra-ui/react'
import CH from 'src/types'

const TESTID_FORM_SKELETON = 'ch-form-skeleton'
const TESTID_FORM_SKELETON_INPUT = 'ch-form-skeleton-input'

const FormSkeleton: React.FC<CH.ClientsLimit> = ({ quantity }) => {
  return (
    <SimpleGrid
      columns={[1, null, 2]}
      spacingX={8}
      spacingY={6}
      data-testid={TESTID_FORM_SKELETON}
    >
      {[...Array(quantity)].map((_, index) => (
        <Skeleton
          height={12}
          key={index}
          data-testid={TESTID_FORM_SKELETON_INPUT}
        />
      ))}
    </SimpleGrid>
  )
}

export default FormSkeleton
export { TESTID_FORM_SKELETON, TESTID_FORM_SKELETON_INPUT }
