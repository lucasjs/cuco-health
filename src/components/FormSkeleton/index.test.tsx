import React from 'react'
import { render, screen } from '@testing-library/react'

import FormSkeleton, {
  TESTID_FORM_SKELETON,
  TESTID_FORM_SKELETON_INPUT
} from '../FormSkeleton'

describe('FormSkeleton', () => {
  test('should render correctly', () => {
    const { container } = render(<FormSkeleton quantity={1} />)

    expect(container).toMatchSnapshot()
  })

  test('when pass quantity 4', () => {
    render(<FormSkeleton quantity={4} />)

    expect(screen.getByTestId(TESTID_FORM_SKELETON)).toBeInTheDocument()
    expect(screen.getAllByTestId(TESTID_FORM_SKELETON_INPUT)[0]).toBeVisible()
    expect(screen.getAllByTestId(TESTID_FORM_SKELETON_INPUT)).toHaveLength(4)
  })
})
