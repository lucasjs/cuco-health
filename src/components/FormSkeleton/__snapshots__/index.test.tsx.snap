// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSkeleton should render correctly 1`] = `
<div>
  <div
    class="css-mydr2m"
    data-testid="ch-form-skeleton"
  >
    <div
      class="chakra-skeleton css-64jeh8"
      data-testid="ch-form-skeleton-input"
    />
  </div>
</div>
`;
